src/app/gene-profiles-table/gene-profiles-table.component.ts

```diff
@@ -207,15 +205,35 @@ export class GeneProfilesTableComponent extends StatefulComponent implements OnI
         this.highlightedGenes = state.highlightedRows;
         this.sortBy = state.sortBy;
         this.orderBy = state.orderBy;
-        this.config = state.config;
+        this.leavesIds = state.headerLeaves;
       });
 
     this.search(this.geneInput);
-    this.isStateLoaded = true;
+  }
+
+  public setIds(): void {
+    if (this.leavesIds.length) {
+      this.leaves = GeneProfilesColumn.allLeaves(this.config.columns);
+      this.leaves.map(leaf => {
+        if (this.leavesIds.includes(leaf.id)) {
+          leaf.visibility = true;
+        } else {
+          leaf.visibility = false;
+        }
+      });
+    }
```

Comment:
I think this method could use a better name - something like `setLeavesVisibility` ?